### PR TITLE
Make AWS IAM resource creation idempotent

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
NONE
```

/assign @mrIncompetent

### CODE REVIEW

File: api/pkg/provider/cloud/aws/provider.go [hunk 1]
```diff
@@ -356,11 +357,23 @@ func createInstanceProfile(client *iam.IAM, name string) (*iam.Role, *iam.Instan
...
+	var role *iam.Role
 	rOut, err := client.CreateRole(paramsRole)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to create role: %v", err)
+		if !isEntityAlreadyExists(err) {
+			return nil, nil, fmt.Errorf("failed to create role: %v", err)
+		}
+		// Accept "EntityAlreadyExists" and assume the config is correct
+		paramsRoleGet := &iam.GetRoleInput{RoleName: roleName}
+		getRoleOut, err := client.GetRole(paramsRoleGet)
```
Comment (mrIncompetent) [inline api/pkg/provider/cloud/aws/provider.go:373]: It feels uncommon to first call `Create` and if it already exists we call `Get`.
Usually, such logic is handled the other way around.
I mean yeah, we save API requests in the long run (If we hit the happy path most of the time)...

This is more a side note and not a request for change

File: api/pkg/provider/cloud/aws/provider.go [hunk 2]
```diff
@@ -374,24 +387,46 @@ func createInstanceProfile(client *iam.IAM, name string) (*iam.Role, *iam.Instan
...
+		instanceProfile = cipOut.InstanceProfile
 	}
 
-	paramsAddRole := &iam.AddRoleToInstanceProfileInput{
-		InstanceProfileName: aws.String(kubermaticInstanceProfileName), // Required
-		RoleName:            aws.String(kubermaticRoleName),            // Required
+	var instanceProfileHasRole bool
+	for _, role := range instanceProfile.Roles {
+		if *role.RoleName == *roleName {
+			instanceProfileHasRole = true
```
Comment (mrIncompetent) [inline api/pkg/provider/cloud/aws/provider.go]: Return to avoid the below `if` statement and `instanceProfileHasRole`

**Comment by alvaroaleman** [comment]
@mrIncompetent Updated, PTAL

**Pull request merged**